src/components/Notifications/GetNotifications.tsx

```diff
@@ -0,0 +1,39 @@
+import { useQuery } from "@tanstack/react-query";
+
+import request from "@/Utils/request/request";
+import { QueryRoute, RequestOptions } from "@/Utils/request/types";
+
+export interface QueryOptions<TData> extends RequestOptions<TData> {
+  prefetch?: boolean;
+  key?: string;
+}
+
+/**
+ * Fetch data using TanStack's useQuery directly
+ */
+export function useGetNotifications<TData>(
+  route: QueryRoute<TData>,
+  options?: QueryOptions<TData>,
+) {
+  const {
+    data: response,
+    error,
+    isFetching: loading,
+    refetch,
+  } = useQuery({
+    queryKey: [route.path, options?.pathParams, options?.query, options?.key],
+    queryFn: async ({ signal }) => {
+      return await request(route, { ...options, signal });
+    },
+    enabled: options?.prefetch ?? true,
+    refetchOnWindowFocus: false,
+  });
+
+  return {
+    data: response?.data,
+    loading,
+    error,
+    res: response?.res,
+    refetch,
+  };
+}
```

Comment:
1. this is not going to get reused in multiple places, so why not inline within the component itself? let's not keep it separate.
2. Notice board and notifications are two different things. Use wordings carefully.
3. `request` is deprecated. Refer docs as mentioned before. use `query`.

Reply:
@rithviknishad Removed the useGetNotifications hook and used in the component and created a new NoticeData for notices and Used query instead of request .